src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -64,6 +65,9 @@ String StorageObjectStorage::getPathSample(StorageInMemoryMetadata metadata, Con
         {} // file_progress_callback
     );
 
+    if (dynamic_cast<StorageObjectStorageSource::KeysIterator *>(file_iterator.get()))
+        return "";
```

Comment:
We should not return empty path in this case, we should return any path from the list of paths stored inside `KeysIterator`.

I think it will be better to do:
```c++
    if (!configuration->isArchive() && !configuration->isPathWithGlobs() && !local_distributed_processing)
        return configuration->getPath();
```
Before creating the iterator instead